lib/pdk/logger.rb

```diff
@@ -29,6 +30,13 @@ def initialize
       self.level = ::Logger::INFO
     end
 
+    def warn_once(*args)
+      hash = args.inspect.hash
+      return if @sent_messages.key?(hash)
+      @sent_messages[hash] = true
```

Comment:
Should we keep track of these per log level? Or does that already get captured by `args.inspect.hash`?

Reply:
If down the line we implement other `*_once` methods then yeah we'll need to track them by severity

Reply:
Pushed a quick update to track by severity